typing/ctype.ml

```diff
@@ -1606,36 +1606,25 @@ let expand_abbrev_gen kind find_type_expansion env ty =
       let level = get_level ty in
       let scope = get_scope ty in
       let lookup_abbrev = proper_abbrevs args abbrev in
-      begin match find_expans kind path !lookup_abbrev with
+      begin try match find_expans kind path !lookup_abbrev with
         Some ty' ->
           (* prerr_endline
             ("found a "^string_of_kind kind^" expansion for "^Path.name path);*)
-          if level <> generic_level then
-            begin try
-              update_level env level ty'
-            with Escape _ ->
-              (* XXX This should not happen.
-                 However, levels are not correctly restored after a
-                 typing error *)
-              ()
-            end;
-          begin try
-            update_scope scope ty';
-          with Escape _ ->
-            (* XXX This should not happen.
-               However, levels are not correctly restored after a
-               typing error *)
-            ()
-          end;
+          if level <> generic_level then update_level env level ty';
+          update_scope scope ty';
           ty'
-      | None ->
-          match find_type_expansion path env with
-          | exception Not_found ->
+      | None -> raise Not_found
```

Comment:
I am not fond of using `Not_found` as a control flow exception here because `Not_found` is raised too often in this part of the typechecker. What do you think of using a result to keep a more linear control flow (something like https://github.com/ocaml/ocaml/commit/040a45b15a6e1f87c03cc0880b7d43a105e53cfe ?).

Reply:
Done. This becomes verbose though.

Reply:
I like this change. Thank you @Octachron .